common/leica_sep.c

```diff
@@ -37,64 +45,104 @@ static int leica_sep_transfer(struct leica_sep_funcs *f, char *cmd, char *data,
 
 	f->puts(cmd);
 	cmd[1] = 'R';
-	*d++ = '-';
 	while(get_timer(0) < (start + timeout_ms))
 	{
 		if (!f->tstc())
 			continue;
 
-		data[bytes_received++] = f->getc();
-		*d++ = data[bytes_received -1];
+		data[bytes_received] = f->getc();
 
-		if ((data[bytes_received - 1] == '\n') || (bytes_received >= size)) {
-			if (bytes_received >= cmd_length &&
-			    !strncmp(data, cmd, cmd_length))
-			{
+		if ((data[bytes_received] == '\n') ||
+		    (bytes_received + 1 >= size)) {
+			if (bytes_received + 1 >= cmd_length &&
+			    !strncmp(data, cmd, cmd_length)) {
```

Comment:
As we are now checking for `':'` we can drop `cmd_length`, right?

Reply:
Actually not, data till ":" can be longer than cmd, because it also includes the CRC this is why. So, we still need cmd_length.

Reply:
Agree